src/asmcnc/apps/systemTools_app/screens/xy_jig/screen_xy_jig.py

```diff
@@ -0,0 +1,566 @@
+                self.m.jog_absolute_single_axis(self.axis, self.max_travel, self.max_speed / 5)
+                Clock.schedule_once(self.record_away_values, 0.4)
+            else:
+                Clock.schedule_once(self.start_moving_away, 0.1)
+
+    def record_away_values(self, dt):
+        if self.test_running:
+            if self.m.state().startswith('Idle'):
+                self.m.jog_absolute_single_axis(self.axis, -1, self.max_speed / 5)
+                Clock.schedule_once(self.record_home_values, 0.4)
+            else:
+                if self.axis == 'Y':
+                    sg_value = self.m.s.sg_y_axis
+                    sg_value_motor_1 = self.m.s.sg_y1_motor
+                    sg_value_motor_2 = self.m.s.sg_y2_motor
+                    pos = self.m.mpos_y()
+                else:
+                    sg_value = self.m.s.sg_x_motor_axis
+                    sg_value_motor_1 = self.m.s.sg_x1_motor
+                    sg_value_motor_2 = self.m.s.sg_x2_motor
+                    pos = self.m.mpos_x()
+
+                if sg_value not in [-999, None]:
+                    self.sg_values_away.append(sg_value)
+                    self.pos_values_away.append(pos)
+
+                if sg_value_motor_1 not in [-999, None]:
+                    self.sg_values_away_motor_1.append(sg_value_motor_1)
+                    self.pos_values_away_motor_1.append(pos)
+
+                if sg_value_motor_2 not in [-999, None]:
+                    self.sg_values_away_motor_2.append(sg_value_motor_2)
+                    self.pos_values_away_motor_2.append(pos)
+
+                Clock.schedule_once(self.record_away_values, 0.1)
+
+    def record_home_values(self, dt):
+        if self.test_running:
+            if self.m.state().startswith('Idle'):
+                PopupPhaseTwo(self.systemtools_sm.sm, self.l)
+            else:
+                if self.axis == 'Y':
+                    sg_value = self.m.s.sg_y_axis
+                    sg_value_motor_1 = self.m.s.sg_y1_motor
+                    sg_value_motor_2 = self.m.s.sg_y2_motor
+                    pos = self.m.mpos_y()
+                else:
+                    sg_value = self.m.s.sg_x_motor_axis
+                    sg_value_motor_1 = self.m.s.sg_x1_motor
+                    sg_value_motor_2 = self.m.s.sg_x2_motor
+                    pos = self.m.mpos_x()
+
+                if sg_value not in [-999, None]:
+                    self.sg_values_home.append(sg_value)
+                    self.pos_values_home.append(pos)
+
+                if sg_value_motor_1 not in [-999, None]:
+                    self.sg_values_home_motor_1.append(sg_value_motor_1)
+                    self.pos_values_home_motor_1.append(pos)
+
+                if sg_value_motor_2 not in [-999, None]:
+                    self.sg_values_home_motor_2.append(sg_value_motor_2)
+                    self.pos_values_home_motor_2.append(pos)
+
+                Clock.schedule_once(self.record_home_values, 0.1)
```

Comment:
That would probably have been the better way to do it, but the `if self.test_running:` statement at the beginning of each function solves this as it automatically stops scheduling any more events, so I'm going to keep this as it is since it seems to be working, if that's alright?